src/Ideas.js

```diff
@@ -8,91 +8,116 @@ import './Ideas.css';
 
 slug.defaults.mode = 'rfc3986';
 
-const defaultSelection = ['Economic Development', 'Data and Expertise', 'Resiliency and Sustainability', 'Neighborhood Improvement', 'Housing'];
+const allCategories = ['Economic Development', 'Data and Expertise', 'Resiliency and Sustainability', 'Neighborhood Improvement', 'Housing'];
 
-function checkIfAllSelected(categories, all) {
+const allValues = {};
+
+const checkIfAllSelected = (categories, all) => {
   let allSelected = true;
   all.forEach((d) => {
     if (categories.indexOf(d) < 0) allSelected = false;
   });
   return allSelected;
-}
+};
 
 // utility helpers for arrays
-function removeItem(array, value) {
+const removeItem = (array, value) => {
   const index = array.indexOf(value);
 
   if (index > -1) {
     array.splice(index, 1);
   }
 
   return array;
-}
+};
+
+const unique = array => array.filter((x, i, a) => a.indexOf(x) === i);
+
 
 class Ideas extends Component {
   constructor(props) {
     super();
-    const { location } = props;
+    const { location, ideas } = props;
     const query = new URLSearchParams(location.search);
-    const value = query.get('categories') || '';
+    const queryCategories = query.get('categories') || '';
+    const queryTags = query.get('tags') || '';
+
+    allValues.categories = allCategories;
+    allValues.tags = unique(ideas.reduce((a, b) => a.concat(b.tags), []).filter(Boolean));
```

Comment:
I know this is from one of my old commits, but I think with function argument names, we should go wild with how they're named (in terms of length). No need to use a,b,c - the minifier in the build step does that for us.